internal/driver/glfw/loop.go

```diff
@@ -149,23 +151,24 @@ func (d *gLDriver) runGL() {
 						view.SetSize(w.shouldWidth, w.shouldHeight)
 					}
 				}
-
 			}
 
 			d.animation.TickAnimations()
 			d.drawSingleFrame()
-		case set := <-settingsChange:
+		}
+
+		if pendingSettings != nil {
 			painter.ClearFontCache()
 			cache.ResetThemeCaches()
-			app.ApplySettingsWithCallback(set, fyne.CurrentApp(), func(w fyne.Window) {
+			app.ApplySettingsWithCallback(pendingSettings, fyne.CurrentApp(), func(w fyne.Window) {
 				c, ok := w.Canvas().(*glCanvas)
 				if !ok {
 					return
 				}
 				c.applyThemeOutOfTreeObjects()
 				c.reloadScale()
 			})
```

Comment:
Is there any specific reason as to why we can't do this in the change listener directly if it already runs on the main thread anyhow?